lib/net.c

```diff
@@ -1200,12 +1201,119 @@ start_iface(struct gatekeeper_if *iface, unsigned int num_attempts_link_get)
 	return ret;
 }
 
+static void
+calculate_net_config_para(struct net_config *net_conf)
+{
+	int i;
+
+	/* The total number of RX queues on any interface. */
+	uint16_t gatekeeper_num_rx_queues = net_conf->front.num_rx_queues +
+		(net_conf->back_iface_enabled ?
+		 net_conf->back.num_rx_queues : 0);
+
+	/* The total number of TX queues on any interface. */
+	uint16_t gatekeeper_num_tx_queues = net_conf->front.num_tx_queues +
+		(net_conf->back_iface_enabled ?
+		 net_conf->back.num_tx_queues : 0);
+
+	const char *filtered_blocks[] = { "dynamic_conf" };
+
+	/*
+	 * The total number of lcores used by any functional block
+	 * for either RX or TX except for Dynamic Configuration.
+	 */
+	int gatekeeper_num_lcores = launch_count_lcores(filtered_blocks,
+		RTE_DIM(filtered_blocks));
+
+	/*
+	 * The maximum number of receive descriptors to
+	 * allocate for the receive ring over all interfaces.
+	 */
+	uint16_t gatekeeper_num_rx_desc =
+		net_conf->front.gatekeeper_num_rx_desc;
+
+	/*
+	 * The maximum number of transmit descriptors to
+	 * allocate for the transmit ring over all interfaces.
+	 */
+	uint16_t gatekeeper_num_tx_desc =
+		net_conf->front.gatekeeper_num_tx_desc;
+
+	if (net_conf->back_iface_enabled) {
+		gatekeeper_num_rx_desc = RTE_MAX(gatekeeper_num_rx_desc,
+			net_conf->back.gatekeeper_num_rx_desc);
+
+		gatekeeper_num_tx_desc = RTE_MAX(gatekeeper_num_tx_desc,
+			net_conf->back.gatekeeper_num_tx_desc);
+	}
+
+	/*
+	 * The number of elements in the mbuf pool.
+	 *
+	 * The optimum size (in terms of memory usage) for a mempool is when
+	 * it is a power of two minus one.
+	 *
+	 * Need to provision enough memory for the worst case. According to
+	 * l3fwd sample application, it's essentially the number of RX
+	 * descriptors (across all queues for all ports), the number of TX
+	 * descriptors (across all queues for all ports), the number of packet
+	 * burst buffers (across all lcores for all bursts), the number of
+	 * slots in the packet buffer cache (across all lcores).
+	 */
+	net_conf->gatekeeper_num_mbuf = rte_align32pow2(
+		gatekeeper_num_rx_queues * gatekeeper_num_rx_desc +
+		net_conf->gatekeeper_num_ports *
+		net_conf->gatekeeper_total_pkt_burst +
+		gatekeeper_num_tx_queues * gatekeeper_num_tx_desc +
+		gatekeeper_num_lcores * RTE_MEMPOOL_CACHE_MAX_SIZE) - 1;
+
+	/*
+	 * XXX #155 The size of the per-core object cache, i.e.,
+	 * number of struct rte_mbuf elements in the per-core object
+	 * cache. This should be analyzed or tested further to find
+	 * optimal value.
+	 *
+	 * Each core deals with at most gatekeeper_num_ports queues,
+	 * so the cache size should be at least
+	 * (number of ports * number of descriptors per queue), i.e.,
+	 * (gatekeeper_num_ports * GATEKEEPER_DESC_PER_QUEUE).
```

Comment:
Drop the paragraph above. It's talking about `net_conf->gatekeeper_num_mbuf` not about `net_conf->gatekeeper_per_core_cache_size`. Moreover, the content of this paragraph is now oudtaded.